pkg/console/util.go

```diff
@@ -277,12 +278,35 @@ func getRemoteSSHKeys(url string) ([]string, error) {
 }
 
 func getFormattedServerURL(addr string) (string, error) {
-	ipErr := checkIP(addr)
-	domainErr := checkDomain(addr)
-	if ipErr != nil && domainErr != nil {
+	if addr == "" {
+		return "", errors.New("management address cannot be empty")
+	}
+	addr = strings.TrimSpace(addr)
+
+	realAddr := addr
+	if !strings.HasPrefix(addr, https) {
+		realAddr = https + addr
+	}
+	parsedUrl, err := url.ParseRequestURI(realAddr)
+	if err != nil {
+		return "", fmt.Errorf("%s is invalid", addr)
+	}
+
+	host := parsedUrl.Hostname()
+	if checkIP(host) != nil && checkDomain(host) != nil {
 		return "", fmt.Errorf("%s is not a valid ip/domain", addr)
 	}
-	return fmt.Sprintf("https://%s:%s", addr, rancherManagementPort), nil
+
+	if parsedUrl.Path != "" {
+		return "", fmt.Errorf("path is not allowed in management address: %s", parsedUrl.Path)
+	}
+
+	port := parsedUrl.Port()
+	if port == "" {
+		parsedUrl.Host += ":443"
+	}
```

Comment:
```go
else if port != "443" {

}
```

Does RKE2 support custom port?
Refer to the logs in https://github.com/harvester/harvester/issues/2737#issue-1362758670
It seems that RKE2 only support 8443 and 443.
And for Harvester, we only use 443.

Reply:
So might be better to add a port verification if not using 443 or 8443?

Reply:
just need make sure the port is 443

Reply:
8443 was close by Harvester for security